include/nn/uds/Cafe.h

```diff
@@ -0,0 +1,471 @@
+#pragma once
+
+#include <wut.h>
+#include <nn/result.h>
+#include <nn/uds/NodeInformation.h>
+#include <nn/cfg/CTR.h>
+
+#ifdef __cplusplus
+
+//! Theoretical supported maximum number of nodes connected to a network, including the host
+#define UDS_MAX_NODES 0x10
+//! Broadcast node ID, used for sending data to all nodes in the network
+#define UDS_BROADCAST_NODE_ID 0xFFFF
+
+namespace nn {
+
+namespace uds {
+
+namespace Cafe {
+   //! Stores the connection status of the network
+   struct ConnectionStatus {
+      //! The network status
+      uint32_t status;
+      //! The reason why the status has changed
+      uint32_t reason;
+      //! The network node ID of this device
+      uint16_t networkNodeId;
+      //! Bitmask of nodes that have changed
+      uint16_t changedNodes;
+      //! Array of nodes connected to the network
+      uint16_t nodes[UDS_MAX_NODES];
+      //! Total amount of nodes connected to the network
+      uint8_t totalNodes;
+      //! Maximum number of nodes that can connect to the network, including the host
+      uint8_t maxNodes;
+      //! Bitmask of nodes connected to the network
+      uint16_t nodeBitmask;
+   };
+   WUT_CHECK_SIZE(ConnectionStatus, 0x30);
+   WUT_CHECK_OFFSET(ConnectionStatus, 0x00, status);
+   WUT_CHECK_OFFSET(ConnectionStatus, 0x04, reason);
+   WUT_CHECK_OFFSET(ConnectionStatus, 0x08, networkNodeId);
+   WUT_CHECK_OFFSET(ConnectionStatus, 0x0A, changedNodes);
+   WUT_CHECK_OFFSET(ConnectionStatus, 0x0C, nodes);
+   WUT_CHECK_OFFSET(ConnectionStatus, 0x2C, totalNodes);
+   WUT_CHECK_OFFSET(ConnectionStatus, 0x2D, maxNodes);
+   WUT_CHECK_OFFSET(ConnectionStatus, 0x2E, nodeBitmask);
+
+   //! Recommended size to use for work memory buffer
+   static constexpr uint32_t kWorkMemorySize = 0x14000;
+
+   //! Describes a network endpoint
+   typedef uint32_t EndpointDescriptor;
```

Comment:
The other C++ only headers in wut use an alias declaration instead of a typedef.
```suggestion
   using EndpointDescriptor = uint32_t;
```